core/tee/tee_svc_storage.c

```diff
@@ -273,19 +277,60 @@ static TEE_Result tee_svc_storage_read_head(struct tee_ta_session *sess,
 static TEE_Result tee_svc_storage_init_file(struct tee_ta_session *sess,
 					    struct tee_obj *o,
 					    struct tee_obj *attr_o, void *data,
-					    uint32_t len, uint32_t flags)
+					    uint32_t len)
 {
 	TEE_Result res = TEE_SUCCESS;
 	int fd;
-	int err;
+	int err = -1;
 	struct tee_svc_storage_head head;
-	char *file = NULL;
+	char *initfile = NULL;
 
 	if (o == NULL || o->pobj == NULL)
 		return TEE_ERROR_BAD_PARAMETERS;
 
 	free(o->data);
 
+	initfile = tee_svc_storage_create_filename(sess,
+						   o->pobj->obj_id,
+						   o->pobj->obj_id_len,
+						   true);
+
+	if (initfile == NULL)
+		return TEE_ERROR_OUT_OF_MEMORY;
+
+	err = -1;
+	err = tee_file_ops.access(initfile, TEE_FS_F_OK);
+	if (!err) {
+		/* initfile exists */
+		err = tee_file_ops.unlink(initfile);
+		if (err) {
+			free(initfile);
+			initfile = NULL;
+			return TEE_ERROR_STORAGE_NOT_AVAILABLE;
+		}
+	}
+
+	res = tee_svc_storage_create_file(sess, initfile, &fd);
+	if (res != TEE_SUCCESS)
+		return res;
+
+	/* error codes needs better granularity */
+	if (fd < 0)
+		return TEE_ERROR_GENERIC;
+
+	err = -1;
+	err = tee_file_ops.access(initfile, TEE_FS_F_OK);
+	if (err) {
+		/* file does not exist */
+		free(initfile);
+		initfile = NULL;
+		res = TEE_ERROR_STORAGE_NOT_AVAILABLE;
+		goto exit;
+	}
+
+	free(initfile);
```

Comment:
free(initfile);
    initfile = NULL;

This can be moved before the `if (err)`